src/test/java/org/zalando/intellij/swagger/completion/SwaggerFixture.java

```diff
@@ -0,0 +1,122 @@
+package org.zalando.intellij.swagger.completion;
+
+import com.intellij.testFramework.LightProjectDescriptor;
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture;
+import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
+import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory;
+import com.intellij.testFramework.fixtures.TestFixtureBuilder;
+import com.intellij.testFramework.fixtures.impl.LightTempDirTestFixtureImpl;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsCollectionContaining;
+import org.hamcrest.core.IsNot;
+import org.hamcrest.core.IsNull;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Assert;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SwaggerFixture {
+    private final CodeInsightTestFixture myCodeInsightFixture;
+
+    @NotNull
+    public static SwaggerFixture forResourceFolder(@NotNull String resourceBasedPath) throws Exception {
+        String absolutePath = toAbsolutePath(resourceBasedPath);
+        return new SwaggerFixture(absolutePath);
+    }
+
+    public SwaggerFixture(@NotNull String folderAbsolutePath) throws Exception {
+        IdeaTestFixtureFactory factory = IdeaTestFixtureFactory.getFixtureFactory();
+        LightProjectDescriptor projectDescriptor = LightProjectDescriptor.EMPTY_PROJECT_DESCRIPTOR;
+        TestFixtureBuilder<IdeaProjectTestFixture> fixtureBuilder = factory.createLightFixtureBuilder(projectDescriptor);
+        IdeaProjectTestFixture fixture = fixtureBuilder.getFixture();
+
+        myCodeInsightFixture = factory.createCodeInsightFixture(fixture, new LightTempDirTestFixtureImpl(true));
+        myCodeInsightFixture.setUp();
+        myCodeInsightFixture.setTestDataPath(folderAbsolutePath);
+    }
+
+    public void tearDown() throws Exception {
+        myCodeInsightFixture.tearDown();
+    }
+
+    @NotNull
+    public AssertableList<String> getCompletions(@NotNull String caretFileName) {
+        List<String> results = myCodeInsightFixture.getCompletionVariants(caretFileName);
+        Assert.assertThat(results, IsNull.notNullValue());
+        return new AssertableList<>(results);
+    }
+
+    @NotNull
+    public AssertableList<String> getCompletions(@NotNull String caretFileNoExt, @NotNull JsonOrYaml fileKind) {
+        String fullName = fileKind.getFileName(caretFileNoExt);
+        return getCompletions(fullName);
+    }
+
+    public enum JsonOrYaml {
+        JSON("json"),
+        YAML("yaml");
+
+        private final String myFileExtension;
+
+        JsonOrYaml(String fileExtension) {
+            myFileExtension = fileExtension;
+        }
+
+        @NotNull
+        public String getFileName(@NotNull String base) {
+            return base.endsWith(".") ? base + myFileExtension : base + "." + myFileExtension;
+        }
+    }
+
+    public static class AssertableList<T> {
```

Comment:
I think generics are not needed here, we are only playing around with strings.

`AssertableList<T>` to `AssertableList` and
`private final List<T> myActualList;` to `private final List<String> myActualList;`